proteuslib/edb/equations/van_t_hoff_alt_form.py

```diff
@@ -0,0 +1,107 @@
+                    " set as ConcentrationForm.molarity or ConcentrationForm.activity.")
+            elif c_form == ConcentrationForm.partialPressure:
+                raise ConfigurationError(
+                    "{} concentration_form configuration argument must be either "
+                    " set as ConcentrationForm.molarity or ConcentrationForm.activity.")
+            else:
+                raise BurntToast(
+                    "{} get_concentration_term received unrecognised "
+                    "ConcentrationForm ({}). This should not happen - please "
+                    "contact the IDAES developers with this bug."
+                    .format(rblock.name, c_form))
+
+            e_units = c_units**order
+
+        # Create a variable for entropy of reaction
+        #   NOTE: We do not need to create a variable for reaction enthalpy
+        #       because that variable already exists from another method
+        rblock.ds_rxn_ref = Var(doc="Specific entropy of reaction at reference state",
+                                units=pyunits.J/pyunits.mol/pyunits.K)
+        set_param_from_config(rblock, param="ds_rxn_ref", config=config)
+
+        # Add a param for reference temperature (not read in from file)
+        rblock.T_eq_ref = Param(
+                doc="Reference temperature for equilibrium constant",
+                units=units["temperature"],
+                initialize=293)
+
+        # Convert the enthalpy value (if necessary)
+        dh_val = pyunits.convert_value(rblock.dh_rxn_ref.value, from_units=pyunits.get_units(rblock.dh_rxn_ref), to_units=pyunits.J/pyunits.mol)
+
+        # Calculate a k_eq at reference temperature in K
+        k_eq = exp(-(dh_val/8.3145/rblock.T_eq_ref.value) + (rblock.ds_rxn_ref.value/8.3145))
+
+        # Add parameter for equilibrium coefficient at reference temperature
+        #   and initialize its value to the above calculated value, which is
+        #   done on a (mol/L) basis for aqueous chemistry
+        rblock.k_eq_ref = Param(
+                doc="Equilibrium constant at reference state",
+                units=e_units,
+                initialize=k_eq)
+
+        # Convert units of k value to what IDAES wants
+        rblock.k_eq_ref.value = pyunits.convert_value(rblock.k_eq_ref.value, from_units=(pyunits.mol/pyunits.L)**order, to_units=e_units)
+
+        # Lastly, convert units for reference temperature back to the expected Units
+        rblock.T_eq_ref.value = pyunits.convert_value(rblock.T_eq_ref.value, from_units=pyunits.K, to_units=units["temperature"])
+
+    @staticmethod
+    def return_expression(b, rblock, r_idx, T):
+        units = rblock.parent_block().get_metadata().derived_units
+
+        return rblock.k_eq_ref * exp(
+            -(b.dh_rxn[r_idx] /
+              pyunits.convert(c.gas_constant,
+                              to_units=units["gas_constant"])) *
+            (1/T - 1/rblock.T_eq_ref))
```

Comment:
Similar to the previous comment. We are moving away from these original custom functions and trying to utilize more of the built-in IDAES functions. The IDAES <code>gibbs_energy</code> function replaces this class. Its import statement is...
<code>from idaes.generic_models.properties.core.reactions.equilibrium_constant import gibbs_energy</code>

Reply:
ok thanks